object_store.go

```diff
@@ -85,18 +88,49 @@ func (x *minioDriver) ListObjects(bucketName string) []Object {
 			continue
 		}
 
-		opts := minio.StatObjectOptions{}
-		objectInfo, err := x.Client.StatObject(bucketName, objectInfo.Key, opts)
+		object, err := x.StatObject(bucketName, objectInfo.Key)
 		if err != nil {
 			logger.WithError(err).Error("minio.StatObject() failed")
 			continue
 		}
 
-		objects = append(objects, Object{
+		objects = append(objects, object)
+	}
+	return objects
+}
+
+func (x *minioDriver) StatObject(bucketName, objectName string) (Object, error) {
+	opts := minio.StatObjectOptions{}
+	objectInfo, err := x.Client.StatObject(bucketName, objectName, opts)
+	if err != nil {
+		return Object{}, err
+	} else {
+		return Object{
 			ContentType: objectInfo.ContentType,
 			Name:        objectInfo.Key,
 			Tags:        Tags(objectInfo.UserMetadata),
-		})
+		}, nil
+	}
+}
+
+const LinkExpiration = 24 * 3600 * time.Second // 1 Day for protect links
+
+func (x *minioDriver) DownloadURL(bucketName string, objectName string) (string, error) {
+	policy, err := x.Client.GetBucketPolicy(bucketName)
+	if err != nil {
+		return "", err
+	}
+
+	var downloadUrl *url.URL
+	switch policy {
+	case "download":
+		downloadUrl, err = url.Parse(fmt.Sprintf("http://localhost:9000/%s/%s", bucketName, objectName))
```

Comment:
This should probably use the the host/port from config.

Reply:
Actually, it can be a relative url

Reply:
Err I mean to say I don't need to put any host there.

Reply:
:laughing: ha I forgot that was for the storage not api